kotlin-result-coroutines/src/commonMain/kotlin/com/github/michaelbull/result/coroutines/binding/SuspendableBinding.kt

```diff
@@ -12,18 +12,23 @@ import kotlin.contracts.contract
 /**
  * Suspending variant of [binding][com.github.michaelbull.result.binding].
  */
-public suspend inline fun <V, E> binding(crossinline block: suspend SuspendableResultBinding<E>.() -> V): Result<V, E> {
+public suspend inline fun <V, E> binding(eagerlyCancel: Boolean = false, crossinline block: suspend SuspendableResultBinding<E>.() -> V): Result<V, E> {
```

Comment:
I'm not a fan of the boolean parameter. I would rather put the new behaviour in a separate method, e.g. `eagerBinding` or something.

Reply:
ye I think that makes sense - it will probably help with testing too